src/Dialect/ONNX/Rewrite.td

```diff
@@ -431,6 +435,18 @@ def CastEliminationPattern : Pat<
 //  (ONNXCastOp (ONNXCastOp $arg, $_), $type),
 //  (ONNXCastOp $arg, $type)>;
 
+//===----------------------------------------------------------------------===//
+// Canonicalization for ONNXCastLikeOp
+//===----------------------------------------------------------------------===//
+
+def CastLikeEliminationPattern:  Pat<
+  // Remove a CastLike pattern if the first input tensor and second input tensor are the same. 
+  (ONNXCastLikeOp $val, $castTo, $saturate),
+  // Remove the castlike.
+  (replaceWithValue $val),
+  // Check that the first input tensor and second input tensor are the same
+  [(HaveSameElementType $val, $castTo)]>;
+
```

Comment:
I would add here (or probably more likely in Rewrite.cpp) a rule that transforms ALL cast like into casts; and thus this rule could be rewritten for the regular cast (if it does not already exists).